### PR TITLE
Restructure docs, part one: nav and landing pages

### LINKED ISSUE #242: Add landing page for ev3dev drivers/other technical topics
The motor/sensor classes should be easy to access; everything else should be demoted.

From ev3dev/ev3dev#734

### LINKED ISSUE #243: Add descriptions of individual tutorial categories and page as a whole
From ev3dev/ev3dev#734

### LINKED ISSUE #244: Revise nav links to prioritize important information
- Home
- Get started
- Documentation (clicking this should lead to the docs landing page)
  - Tutorials
  - Libraries
  - Low-level driver access
  - More... (also links to landing page)
	- _the landing page needs to have links to everything that is in the Documentation nav menu, at a minimum._
- Get help
- News
- Community
  - Project showcase
  - Contribute
  - Share

From ev3dev/ev3dev#734

### LINKED ISSUE #246: Add a networking landing page and make sure users go through it
From ev3dev/ev3dev#734.

### PR DESCRIPTION
**Pull request by WasabiFan**
- Update nav structure
- Add networking landing page
- Add low-level driver access page
  - Pretty basic; would ideally be expanded on later.
- Re-did docs landing page
- Added descriptions and proper sort to tutorials page
- Fixed misc small style issues

Live at <http://wasabifan.github.io/ev3dev.github.io/>.

Fixes #246
Fixes #244
Fixes #243
Fixes #242

### CODE REVIEW

**Comment by WasabiFan** [comment]
I think I need to add the networking landing page to the documentation nav item.

**Comment by WasabiFan** [comment]
I also need to mark on the "networking" page which tethering options are supported on what platforms. And probably make the warnings bold.

**Comment by WasabiFan** [comment]
Oh, and I need to update Getting Started for these new pages (mostly networking).

File: _includes/header.html [hunk 1]
```diff
@@ -16,24 +16,24 @@
...
                                 <a href="/docs/tutorials">Tutorials</a>
                             </li>
                             <li role="separator" class="divider"></li>
                             <li>
-                                <a href="/docs/motors">Motors</a>
+                                <a href="/docs/libraries">Libraries</a>
                             </li>
                             <li>
-                                <a href="/docs/sensors">Sensors</a>
+                                <a href="/docs/driver-access">Low-level driver access</a>
```
Comment (dlech) [inline _includes/header.html]: I think I mentioned this elsewhere (I'm not reviewing in linear order :smile:), but...

I think "Hardware Drivers" would be simpler.

File: _data/tutorial-groups.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+- id: networking
```
Comment (dlech) [inline _data/tutorial-groups.yml]: We are getting enough networking pages that we might consider splitting them into "Basic Networking" for things like connecting to the Internet and "Advanced Networking" for things like NFS and MQTT.

File: _data/tutorial-groups.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+- id: system
+  title: "Ev3dev System"
+  description: "Configuring, updating and modifying an ev3dev installation."
+- id: website
+  title: This Website
+  description: "Using this website and contributing content."
+- id: hardware-motors
+  title: "Using Motors"
+  description: "Configuring and using motors on ev3dev, from the basic tacho motors to more complex third-party ones."
+- id: hardware-extras
```
Comment (dlech) [inline _data/tutorial-groups.yml]: This group currently has some standard items, like EV3 buttons and LCD.

File: _data/tutorial-groups.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  description: "Configuring a development environment, installing new languages, and using advanced language tools."
+- id: software-editors
+  title: "Text Editors"
+  description: "Tips on using text editors which ship with ev3dev out-of-the-box."
+- id: cross-compiling
+  title: Cross-Compiling Code
+  description: "Compiling code to run on an EV3 or other ev3dev device from a host PC."
+- id: sd-card-image
+  title: SD Card Imaging
+  description: "Deprecated guides on flashing an SD card with ev3dev images."
```
Comment (dlech) [inline _data/tutorial-groups.yml]: Note: This comment is about the whole file, not this line.

Now that we have this information extracted, we could organize this even more by grouping ids into broader categories, like software, hardware and system administration. As it is, the tutorial page is kind of randomly grouped, which makes it hard to find anything.

File: docs/driver-access.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
+---
+title: Low-level driver access
+subtitle: Controlling ev3dev devices through the low-level drivers
```
Comment (dlech) [inline docs/driver-access.md]: This page is not so pretty on the screen. I'm guessing it doesn't get much traffic though, so probably not such a big deal. I do refer to the sensors page quite often though. As it is, the link doesn't really jump out at you, which slows me down a bit.

File: docs/driver-access.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
+---
+title: Low-level driver access
```
Comment (dlech) [inline docs/driver-access.md]: Having "Low-level" in the title and subtitle is redundant. I like "Hardware Drivers" for a title.

Comment (dlech) [inline docs/driver-access.md]: Also `docs/drivers.md` would be a more generic (future-proof) file name.

File: docs/index.html [hunk 7]
```diff
@@ -7,56 +7,49 @@
...
+            have not been migrated from our old wiki yet. We'd love some help
+            with getting these last few pages migrated! If you would like to
+            help out, read more about what's needed in the
+            <a href="../contribute#write-some-documentation">documentation section</a>
             of our "contributing" page.
         </p>
     </div>
 </div>
 
 <h2>System setup</h2>
```
Comment (dlech) [inline docs/index.html]: "Installation and Setup" might be a better title.

File: docs/index.html [hunk 8]
```diff
@@ -7,56 +7,49 @@
...
-        <a href="sensors">Supported Sensors and Other Input Devices</a>
-    </li>
-    <li>
-        <a href="motors">Supported Motors and Other Output Devices</a>
-    </li>
-    <li>
-        <a href="drivers">ev3dev kernel drivers</a>
-    </li>
-</ul>
+<h3>Low-level driver access</h3>
```
Comment (dlech) [inline docs/index.html]: Everything else on this page is `<h2>`

File: _includes/header.html [hunk 9]
```diff
@@ -42,7 +42,10 @@
                         </ul>
                     </li>
                     <li>
-                        <a href="/news" title="News about ev3dev">News</a>
+                        <a href="/support" title="Get Help">News</a>
```
Comment (dlech) [inline _includes/header.html]: This should not say "News"

File: _includes/header.html [hunk 10]
```diff
@@ -51,22 +54,16 @@
                         </a>
                         <ul class="dropdown-menu">
                             <li>
-                                <a href="/projects" title="Discover projects that use ev3dev">Projects</a>
-                            </li>
-                            <li>
-                                <a href="/share" title="Share projects that use ev3dev">Share</a>
+                                <a href="/projects" title="Discover projects that use ev3dev">Project showcase</a>
```
Comment (dlech) [inline _includes/header.html]: For some reason, I feel like this should be the last item on the list (and Contribute should be first). Just intuition I gues.

File: docs/networking.md [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+on your EV3, you'll need to connect to a network. Here are some resources to
+help you out.
+
+# Connecting to the internet
+
+You have a few options for connecting to the internet.
+
+### With a Wi-Fi dongle
+
+The simplest route is to purchase a Wi-Fi dongle and plug it into the EV3's USB
```
Comment (WasabiFan) [inline docs/networking.md]: _If you have a Wi-Fi dongle_ (e.g. the LEGO-compatible one), I think Wi-Fi is the simplest. If you don't, USB will be the easiest as you mentioned. The reason that I put Wi-Fi first was because Windows has issues with USB sharing (you frequently have to disable and then re-enable the option).

How does this sound: I reword and restructure this so that it says something to that effect. If you have a Wi-Fi dongle, use it; otherwise use USB.

File: docs/networking.md [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+port. Almost any USB Wi-Fi dongle which supports Linux will work, including the
```
Comment (WasabiFan) [inline docs/networking.md]: Can you point me to a list of supported ones? I know of the Netgear one, but I was unaware that there were others that they supported. And are there any other dongles that you think are common and known to work that I should include?

File: docs/networking.md [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+one that is supported by the official LEGO software. Once you have plugged in a
+dongle, you can use Brickman's "**_Wireless and Networks_ > _Wi-Fi_**" menu to
+connect to a network.
+
+### With a USB Ethernet adapter
```
Comment (WasabiFan) [inline docs/networking.md]: I agree that it should be below most of the other options, however I think it should still be above the tethering ones, as they're not helpful for most cases. Inevitably we're going to tell people to upgrade something, so I'd rather have them using a method which supplies internet access than have to instruct them to change their set-up and then try again. What do you think about putting it after the USB and Bluetooth internet ones but before the tethering options?

File: docs/networking.md [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+
+If you have a USB Ethernet adapter (or can buy one) which supports Linux, you
+can use it to access the network from your EV3. Just plug it into the brick and
+connect the adapter to an ethernet cable; no configuration is related.
```
Comment (WasabiFan) [inline docs/networking.md]: Damn autocorrect :laughing: I enabled a spellcheck extension for editing Markdown on the site, but apparently it got a bit overzealous after I mistyped something. Thanks for noticing!

File: docs/networking.md [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+tutorial.
+
+### With Bluetooth through a PC
+
+If you have a PC or Android device with Bluetooth capability, you can connect
+your EV3 through it to access the Web. To learn more about this option, check
+out our [Connecting to the Internet via Bluetooth](/docs/tutorials/connecting-to-the-internet-via-bluetooth/)
+tutorial.
+
+# Tethering to a host PC (no internet access)
```
Comment (WasabiFan) [inline docs/networking.md]: That's a widely debated topic :wink: I generally flip-flop between them based on my mood, the time of day, and the distance between my finger and <kbd>Shift</kbd>. I'll change it to match that preference.

File: docs/networking.md [hunk 16]
```diff
@@ -0,0 +1,69 @@
...
+
+If you don't need to connect your brick to the Internet, you can use tether it
```
Comment (WasabiFan) [inline docs/networking.md]: I guess that's what happens when you fuse two sentences together half-way through writing them :laughing:

File: docs/index.html [hunk 17]
```diff
@@ -7,56 +7,49 @@
 <div class="panel panel-info">
     <div class="panel-heading">
         {% include icon.html type="info" %}
-        ev3dev is a community-maintained project that relies on people's contributions.
+        ev3dev is a community-maintained project that relies on community contributions.
```
Comment (WasabiFan) [inline docs/index.html]: Ahhh good idea! I spent more time that I am willing to admit attempting to find a better way to phrase this.

File: docs/index.html [hunk 18]
```diff
@@ -7,56 +7,49 @@
...
-            topic that you think we should have docs for, please consider
-            <a href="../contribute">contributing</a>.
+            If there's a topic that you think we should have docs for, please
+            consider <a href="../contribute">contributing</a>.
         </p>
         <p>
             <a href="https://github.com/ev3dev/ev3dev/issues/153#issuecomment-93602107">Some pages</a>
-            have not been migrated from our old wiki yet. We'd love some help with getting these last few pages migrated:
-            if you would like to help out, read more about what's needed in the <a href="../contribute#write-some-documentation">documentation section</a>
+            have not been migrated from our old wiki yet. We'd love some help
```
Comment (WasabiFan) [inline docs/index.html]: I like the idea of some variety of reward, but I think it opens up some problems; namely, "Who supplies the reward?" and "How do we determine if someone gets it?"

I might be over-thinking this, though. We could probably do something simple and cheap with rewards like that and get reasonable results. Or maybe Bountysource does something to deal with this problem. We'd just need to make sure we were being fair and that we don't get into grey area if someone contributes lackluster content.

Either way, I agree that we should do _something_. Having this alert here is less than ideal if it isn't serving a useful purpose. _Although, I have to say, it does visually break up the page a bit._

File: docs/index.html (hunk 8 above)
Comment (WasabiFan) [inline docs/index.html]: I was imagining this nested under "writing some software." Do you think it should be structured differently?

File: _includes/imaging-tutorial-etcher-banner.html [hunk 19]
```diff
@@ -1,5 +1,6 @@
 {% include icon.html type="info" %}
 There's an easier way to image your SD card! Check out
 [the "Getting Started" guide](/docs/getting-started#step-2-copy-the-image-on-to-the-sd-card){: .alert-link }
-for a cross-platform guide.
+for a cross-platform guide. This tutorial isn't being maintained and may not
```
Comment (WasabiFan) [inline _includes/imaging-tutorial-etcher-banner.html]: We no longer supply `.zip`s, so the Win32DiskImager tutorial is broken (others may also have issues with that -- I can't remember). I played around with having a separate "danger" banner on that one page, but it looked distracting and awkward. Overall my thinking was essentially that we hadn't been giving these pages much attention so I would rather not confuse people; if you would like to continue supporting them, I can rejigger this but we will need to take the Windows one into consideration.

Statistics-wise, it looks like many people find the Ubuntu guide from outside sources; I suspect that Google has picked it up and people are following it for images other than ev3dev as well. Past that, I don't have much data to go off of right now (or the know-how to persuade Google to give it to me) so I can't make a data-driven call on numbers of people choosing these guides over the Etcher option.

File: docs/driver-access.md (hunk 6 above)
Comment (WasabiFan) [inline docs/driver-access.md]: `/docs/drivers` is the path to our index of drivers (literally a list of driver pages). I think the "driver index page" is an important page to have, but I didn't want to have that list thrown in with this content, so I attempted to come up with a creative title and URL for this one. How do you think we should handle this?

File: docs/driver-access.md (hunk 5 above)
Comment (WasabiFan) [inline docs/driver-access.md]: Yeah, this page looks pretty empty. My thought process was essentially the same, because I doubt that it will be getting many page views, and the people viewing it will likely know what they are looking for anyway. Which link are you referring to?

File: _includes/header.html [hunk 20]
```diff
@@ -16,24 +16,24 @@
                     <li>
                         <a href="/" title="There's no place like Home...">Home</a>
                     </li>
+                    <li>
+                        <a href="/docs/getting-started" title="Our Getting Started guide">Get Started</a>
```
Comment (WasabiFan) [inline _includes/header.html]: > I'm guessing that you changed this to "Get Started" to make it shorter. 

Yeah, that and more declarative as opposed to passive.

> That's fine, but it just strikes me as unusual or awkward for some reason. Not sure how to explain it exactly - it's probably just me. Saying "Start Here" seems a bit more natural to me though.

I get what you're trying to say, and I feel the same. I like the "start here" idea.

> Another thing to consider is that most sites that offer software have a "Download" or "Install" link. I know when I am visiting a new software site, this is what I am looking for.

I've been contemplating the same problem. I expect the lack of a "download" button is disconcerting to most users.

> What do you think about creating a new page just for downloads? ...

I like the idea as you describe it. That would also minimize people downloading the wrong image (which, despite my best efforts with button size and placement, still happens). The full form of that page does depend on a finalized whole-page platform selection mechanism, as you mentioned, so we might need to start this without the nice links and then build on it later.

Do you think I should add this now for the current PR in a partial form, or wait until after I get the rest of it restructured and I can come back to platform selection?

File: _includes/header.html [hunk 21]
```diff
@@ -16,24 +16,24 @@
...
                             <li>
-                                <a href="/docs/getting-started">Getting Started</a>
-                            </li>
-                            <li>
                                 <a href="/docs/tutorials">Tutorials</a>
                             </li>
                             <li role="separator" class="divider"></li>
                             <li>
-                                <a href="/docs/motors">Motors</a>
+                                <a href="/docs/libraries">Libraries</a>
```
Comment (WasabiFan) [inline _includes/header.html]: I'd support changing that page to "programming languages". I'll rename it and add a redirect.

File: _includes/header.html (hunk 1 above)
Comment (WasabiFan) [inline _includes/header.html]: Yeah, I can't remember at this point either, but I think we do have a comment for this somewhere else. I'll change this to match whatever we do over there.

File: _includes/header.html (hunk 9 above)
Comment (WasabiFan) [inline _includes/header.html]: I'm not sure what's more worrying: the fact that I made the mistake in the first place, or the fact that I have been staring at a header with two "news" items for the past week without noticing :blush:

File: _includes/header.html (hunk 10 above)
Comment (WasabiFan) [inline _includes/header.html]: I think this depends on what the "projects" page becomes. If we expand it to be a full-fledged "examples" hub, the link to that (which may be the same page or different) should definitely be obvious. As it is now though, I'll flip it around.

File: _data/tutorial-groups.yml (hunk 3 above)
Comment (WasabiFan) [inline _data/tutorial-groups.yml]: How do you think I should split it?

File: _data/tutorial-groups.yml (hunk 4 above)
Comment (WasabiFan) [inline _data/tutorial-groups.yml]: One of my motivations for moving this info into the data file was to give the items a definitive order, so that now the order in the file is the order on the page (that's why it is an array instead of a dictionary). I should be able to break that up one step further to have the meta-groups with ordered items within each.

File: docs/networking.md (hunk 11 above)
Comment (dlech) [inline docs/networking.md]: Hmm... I wasn't considering the Windows ICS issues.

My personal preference would be to connect by USB first then use `connmanctl` to enter my wifi password because it is insanely long. But I imagine I am the exception here.

I'll let you make the call here.

File: docs/networking.md (hunk 12 above)
Comment (dlech) [inline docs/networking.md]: The "officially supported" ones are going to be the ones I actually own.

    0846:9030 NetGear, Inc. WNA1100 Wireless-N 150 [Atheros AR9271]
    7392:7811 Edimax Technology Co., Ltd EW-7811Un 802.11n Wireless Adapter [Realtek RTL8188CUS]
    0bda:8176 Realtek Semiconductor Corp. RTL8188CUS 802.11n WLAN Adapter

The first two are the ones officially supported by LEGO.

The last one is [this](http://www.edupwireless.com/product-1-2-5-mini-wireless-usb-adapter-en/137318) except mine has Dexter Industries branding. It works with the official EV3 firmware as well.

File: docs/networking.md (hunk 13 above)
Comment (dlech) [inline docs/networking.md]: Yes, I meant before tethering. Just last under this h1 heading.

File: docs/networking.md (hunk 12 above)
Comment (WasabiFan) [inline docs/networking.md]: Sounds good; I'll add that.

File: docs/networking.md (hunk 15 above)
Comment (dlech) [inline docs/networking.md]: Well pick one and stick to it. you capitalized it immediately below. I tend to capitalize it because that is what the spell check tells me to do. (Not a very good reason to be sure.)

File: docs/networking.md [hunk 22]
```diff
@@ -0,0 +1,69 @@
...
+with USB, try our [Using USB Tethering](/docs/tutorials/using-usb-tethering/)
+tutorial.
+
+# Further reading
+
+### Remote terminal with SSH
+
+Once you have a connection between your PC and ev3dev, you will almost certainly
+want to establish an SSH connection to the EV3. This allows you to install
+software, run code, and edit files. Read more in [Our nonexistent guide](/docs/tutorials/connecting-to-ev3dev-with-ssh/).
```
Comment (dlech) [inline docs/networking.md]: No, we don't have one. It would be nice if we did though.

File: docs/index.html (hunk 18 above)
Comment (dlech) [inline docs/index.html]: Actually, ev3dev stickers might be good reward.

We don't have to figure this out right now though.

File: docs/index.html (hunk 8 above)
Comment (dlech) [inline docs/index.html]: I don't think of hardware drivers as writing software. I refer to those pages often just to get technical info on sensors.

File: docs/index.html (hunk 18 above)
Comment (WasabiFan) [inline docs/index.html]: > Actually, ev3dev stickers might be good reward.

Oh, good idea! I recall Ralph offered them to me on multiple occasions but there was an unfortunate disconnect between me putting in on my to-do list and it actually getting done :laughing: Something as simple as stickers like that would be a nice incentive while still only costing a minimal amount.

File: _includes/imaging-tutorial-etcher-banner.html (hunk 19 above)
Comment (dlech) [inline _includes/imaging-tutorial-etcher-banner.html]: I may actually do zips for the next release to take advantage of the etcher metadata feature (unless etcher releases tar/metadata support before we release the image).

Either way, since the statement "This tutorial isn't being maintained and may not..." is in the banner, it leads me to believe that "This" is the etcher tutorial mentioned in the same box rather than the Win32DiskImager tutorial.

IMHO, if we aren't going to maintain a page, then we should remove it. I would say fix the page if some step is wrong for now and we can remove the page when we are confident that etcher works for everyone.

File: docs/driver-access.md (hunk 6 above)
Comment (dlech) [inline docs/driver-access.md]: `/docs/drivers/overview` maybe?

Comment (dlech) [inline docs/driver-access.md]: or `/docs/driver-overview`

File: docs/driver-access.md (hunk 5 above)
Comment (dlech) [inline docs/driver-access.md]: The sensor page is the one I use the most.

File: _includes/header.html (hunk 20 above)
Comment (dlech) [inline _includes/header.html]: let's do the downloads page in a separate PR. there are enough changes in this one already!

File: _data/tutorial-groups.yml (hunk 3 above)
Comment (dlech) [inline _data/tutorial-groups.yml]: Maybe combine it with motors and change it to `hardware-standard`?

File: _includes/imaging-tutorial-etcher-banner.html (hunk 19 above)
Comment (WasabiFan) [inline _includes/imaging-tutorial-etcher-banner.html]: OK, I'll update it to include instructions to download the release from GitHub and assume there's a .zip download available.

File: _data/tutorial-groups.yml (hunk 3 above)
Comment (WasabiFan) [inline _data/tutorial-groups.yml]: Is the "mindsensors.com NxtMMX Motor Controller" considered standard, or should I put it in the "extras" category?

Comment (dlech) [inline _data/tutorial-groups.yml]: It depends. Does `hardware-extra` mean hardware that was not necessarily designed to be LEGO compatible? If so, the NxtMXX would be standard since it is designed for LEGO.

File: docs/tutorials/index.html [hunk 23]
```diff
@@ -4,30 +4,35 @@
 include_masonry: "true"
 ---
 
-{% assign tutorials=site.pages | where: "category", "tutorials" | sort: "subject" | group_by: "subject" %}
+<p class="lead">
+    These tutorials will walk you through working with a wide range of hardware and
+    software. If you're looking for help with a specific goal, this is the place to
+    look!
```
Comment (bmegli) [inline docs/tutorials/index.html]: Are tutorials meant only for hardware and software or are other concepts welcome? 
This dertermines also what people will think they can contribute.

I mean for example:
Robotics - Odometry
Robotics - Motion Control
Kalman Filters, SLAM, etc.


E.g. with ev3dev example realization

File: docs/networking.md (hunk 11 above)
Comment (bmegli) [inline docs/networking.md]: I would second @WasabiFan here.

Whatever you do you will likely endup using Wi-Fi.  Or just most people will.

Other means of connection are great if one can't use Wi-Fi for some reason (USB host used by something else, no dongle yet, no driver for dongle, etc).

File: docs/tutorials/index.html (hunk 23 above)
Comment (WasabiFan) [inline docs/tutorials/index.html]: I think that the tutorials should generally be focused on technical things that one might want to do with ev3dev which are specific to our platform. In general, I want to keep the number of "novel but tangential" tutorials to a minimum. It's a fine line, though; for example, I really like the LIDAR and RasPi camera ones, but I would rather not have generic "robotics" tutorials like odometry as a whole. That type of thing is probably better suited for a project page.

File: _includes/imaging-tutorial-etcher-banner.html (hunk 19 above)
Comment (WasabiFan) [inline _includes/imaging-tutorial-etcher-banner.html]: Note to self: Remove "deprecated" from tutorial category description

File: docs/tutorials/nxtmmx.md [hunk 24]
```diff
@@ -1,6 +1,6 @@
 ---
 title: Using the mindsensors.com NxtMMX Motor Controller
-subject: Hardware - Motors
+subject: hardware-motors
```
Comment (WasabiFan) [inline docs/tutorials/nxtmmx.md]: Done

File: docs/driver-access.md (hunk 5 above)
Comment (WasabiFan) [inline docs/driver-access.md]: What do you think I should do here? Emphasize the links in each bullet?

File: docs/index.html (hunk 18 above)
Comment (WasabiFan) [inline docs/index.html]: Is there anything you want me to do for this right now in this PR?

**Comment by WasabiFan** [comment]
I think I've updated everything based on feedback, aside from two things which I commented on (I wasn't sure if there were any action items from those two). Preview site updated as well.

File: docs/tutorials/connecting-to-ev3dev-with-ssh.md [hunk 25]
```diff
@@ -1,6 +1,6 @@
 ---
 title: Connecting to Ev3dev Using SSH
```
Comment (bmegli) [inline docs/tutorials/connecting-to-ev3dev-with-ssh.md]: I know the reason but this capitalization looks strange (Ev3dev) 

Shouldn't it be ev3dev instead?

File: _data/tutorial-groups.yml [hunk 26]
```diff
@@ -1,27 +1,39 @@
...
+    title: Extra Hardware
+    description: "Configuration and usage of specialty and otherwise nonstandard hardware."
+- title: "System administration & setup"
+  id: "administration-and-setup"
+  items:
+  - id: sd-card-image
+    title: SD Card Imaging
+    description: "Guides on flashing an SD card with ev3dev images."
+  - id: system
+    title: "Ev3dev System"
```
Comment (bmegli) [inline _data/tutorial-groups.yml]: Hmm, capitalization looks strange.

It seems like the small letter became part of the name but maybe that's just for me.

File: docs/tutorials/upgrading-ev3dev.md [hunk 27]
```diff
@@ -1,6 +1,6 @@
 ---
 title: Upgrading Ev3dev
```
Comment (bmegli) [inline docs/tutorials/upgrading-ev3dev.md]: again, I understand the reason of capitalization... 

Again the same feeling that it should be ev3dev

File: _data/tutorial-groups.yml (hunk 26 above)
Comment (bmegli) [inline _data/tutorial-groups.yml]: Also "Step 2: Copy the image on to the SD card"

Shouldn't it be  "Step 2: Copy the image to the SD card" ?

This was not introduced in this commit howerver.

Comment (bmegli) [inline _data/tutorial-groups.yml]: Another one to clarify:

"For Raspberry Pi you will need to use the wired Ethernet connection first."

Should probably be:

"For Raspberry Pi you will need to use the wired Ethernet connection or a display & keyboard first."

Raspberry PIs have video output port so one can use display and keyboard instead of Ethernet.

File: docs/networking.md [hunk 28]
```diff
@@ -13,17 +13,23 @@ You have a few options for connecting to the internet.
...
-one that is supported by the official LEGO software. Once you have plugged in a
-dongle, you can use Brickman's "**_Wireless and Networks_ > _Wi-Fi_**" menu to
-connect to a network.
-
-### With a USB Ethernet adapter
-
-If you have a USB Ethernet adapter (or can buy one) which supports Linux, you
-can use it to access the network from your EV3. Just plug it into the brick and
-connect the adapter to an ethernet cable; no configuration is related.
+The simplest route is to use a Wi-Fi dongle connected through the EV3's USB
```
Comment (bmegli) [inline docs/networking.md]: How about: "The simplest route is to use a USB Wi-Fi dongle"

Anyway, all of that assumes that poeple have some acces point/router with AP available.

**Comment by bmegli** [comment]
Some things that need fixing but that were not introduced in this pull (I list them to rember):

--------------------------------------------------------

"Step 2: Copy the image on to the SD card"

Shouldn't it be "Step 2: Copy the image to the SD card" ?

-------------------------------------

"For Raspberry Pi you will need to use the wired Ethernet connection first."

Should probably be:

"For Raspberry Pi you will need to use the wired Ethernet connection or a display & keyboard first."

------------------------------------------------

"All not saved data may be lost" (sounds awkward)

Should probably be:

"The data that was not saved may be lost"

-----------------------------------------------------------------------------

", but way easier to enter you passphrase this way"

Should probably be:

", but it's way easier to enter you passphrase this way"

In my language it's ok to not use verbs (we have implicit verb construct) but they taught me that it's not ok in English ;-) (yes, I am also doing that in English...)

File: docs/networking.md (hunk 28 above)
Comment (bmegli) [inline docs/networking.md]: So how about:

If you have access to the Wi-Fi network the simplest root is to use a USB Wi-Fi dongle.

File: docs/tutorials/connecting-to-ev3dev-with-ssh.md (hunk 25 above)
Comment (WasabiFan) [inline docs/tutorials/connecting-to-ev3dev-with-ssh.md]: This is the "new grammar" as I have taken to calling it :laughing: The issue is that we have a tradeoff between not capitalizing it (even though it is title case) and capitalizing it weirdly as you pointed out. So, we're looking at this:

### Doing Stuff with ev3dev
### Doing Stuff with Ev3dev

Both look wrong. One is more grammatical, the other may look better. We opted to standardize on the latter option, but honestly I don't have much of a preference either way.

File: docs/networking.md (hunk 28 above)
Comment (WasabiFan) [inline docs/networking.md]: Good idea, I'll change that sentence.

**Comment by WasabiFan** [comment]
> Shouldn't it be "Step 2: Copy the image to the SD card" ?

Honestly, I'd prefer "Step 2: Flash the SD card with the downloaded image".

> Should probably be: "For Raspberry Pi you will need to use the wired Ethernet connection or a display & keyboard first."

Assuming that's technically correct (which I believe it is), I like it.

> Should probably be: "The data that was not saved may be lost"

I like "All unsaved data will be lost", although depending on where you are seeing this, it may be better as "All data that you have not backed up to another location will be lost".

> Should probably be: ", but it's way easier to enter you passphrase this way"

Good catch, although we should also fix the "you": it should be "your".

**Comment by WasabiFan** [comment]
> In my language it's ok to not use verbs (we have implicit verb construct) but they taught me that it's not ok in English ;-) 

I have to say, I can't imagine how that would work :confused: Either way, you're correct in that it's not applicable in English.

File: docs/tutorials/connecting-to-ev3dev-with-ssh.md (hunk 25 above)
Comment (ddemidov) [inline docs/tutorials/connecting-to-ev3dev-with-ssh.md]: The logo on the site seems suggest **Doing Stuff with EV3DEV**

**Comment by dlech** [comment]
> Honestly, I'd prefer "Step 2: Flash the SD card with the downloaded image".

Keep it short. "Flash the SD card" will do.

> Assuming that's technically correct (which I believe it is), I like it.

If you want to get technical, it would be a device without a display, so BrickPi on RPi. PiStorms on RPi has a display.


> I like "All unsaved data will be lost", although depending on where you are seeing this, it may be better as "All data that you have not backed up to another location will be lost".

+1 for using the word "backup"

File: docs/tutorials/connecting-to-ev3dev-with-ssh.md (hunk 25 above)
Comment (WasabiFan) [inline docs/tutorials/connecting-to-ev3dev-with-ssh.md]: I think that's a stylistic choice for the logo instead of a grammatical one. Personally, the only potential options that I think are explicitly _incorrect_ are EV3DEV and EV3dev.

Comment (ddemidov) [inline docs/tutorials/connecting-to-ev3dev-with-ssh.md]: [Smallcaps](https://en.wikipedia.org/wiki/Small_caps#Unicode)? ᴇᴠ3ᴅᴇᴠ :smiley_cat:

Comment (WasabiFan) [inline docs/tutorials/connecting-to-ev3dev-with-ssh.md]: :confounded: This confuses my brain

**Comment by bmegli** [comment]
I mean that sentence from the docs:
"For Raspberry Pi you will need to use the wired Ethernet connection first"

> If you want to get technical, it would be a device without a display, so BrickPi on RPi. PiStorms on RPi has a display.

It's in the context of Get Started. All Raspberry PI versions have video output. Technically speaking user doesn't have to use Ethernet. He can plug the display/keyboard and configure whaterver network connection he wants (e.g. WiFi, bluetooth).

> [...] implicit verbs [...]

Hmm, I messed it. We have implicit subject that can be reconstructed from the verb form.

**Comment by dlech** [comment]
> All Raspberry PI versions have video output.

And when you use PiStorms, you will not get a terminal on the built-in video of the RPi by default, so it is not much use.

**Comment by bmegli** [comment]
@dlech written:
> And when you use PiStorms, you will not get a terminal on the built-in video of the RPi by default, so it is not much use

Uhmmmm, agreeed but...:  ;-)

In the docs we have:

> ***For Raspberry Pi you will need to use the wired Ethernet connection first.*** With PiStorms the EV3 options will work too since you have a screen to configure them on.

Which is not true. For RPi/BrickPi you can use screen/keyboard to setup any connection you like without ethernet connection (Wi-Fi, bluetooth, XBee, ZigBee, whatever)

So something along the line is technically correct:

For RasberryPi/BrickPi you can use wired connection or setup other  connection using screen and keyboard. With PiStorms the EV3 options will work too since you have a built-in screen to configure them on.

**Comment by bmegli** [comment]
Or:

For RasberryPi you can use wired Ethernet connection first. With PiStorms the EV3 options will work too since you have a built-in screen to configure them on. With BrickPi you can also setup other connection from the console using external screen and keyboard.

**Comment by WasabiFan** [comment]
@bmegli I think I understand your point, and I agree; if you'd like to open a PR with that extended sentence I think it would make it clearer and we could merge it from there.

@dlech In case it was lost in the discussion, I updated the PR with fixes for the issues you pointed out. There were two that I was unsure on (they appeared to be planning instead of changes) which I didn't handle.

File: docs/getting-started.md [hunk 29]
```diff
@@ -130,7 +130,7 @@ Now it's time to write the image to the card.
 <div class="alert alert-info" markdown="1">
 {% include icon.html type="info" %}
 For more detailed information and more alternatives, check out
-[our other tutorials](/docs/tutorials/#group-sd-card-image){: .alert-link }.
+[our other tutorials](/docs/tutorials/#group-administration-and-setup-sd-card-image){: .alert-link }.
```
Comment (dlech) [inline docs/getting-started.md]: I think you forgot to delete the `-sd-card-image` part.

File: docs/networking.md [hunk 30]
```diff
@@ -0,0 +1,83 @@
+---
+title: Networking
+subtitle: Connecting ev3dev to the internet and other devices
```
Comment (dlech) [inline docs/networking.md]: This should be Title Case.

File: docs/driver-overview.md [hunk 31]
```diff
@@ -0,0 +1,26 @@
+---
+title: Hardware Drivers
+subtitle: Controlling ev3dev devices through the low-level driver APIs
```
Comment (dlech) [inline docs/driver-overview.md]: Subtitle should be Title Case. (It is still a title after all :wink:)

File: docs/networking.md [hunk 32]
```diff
@@ -0,0 +1,83 @@
...
+---
+
+If you want to access your ev3dev installation from a PC or install new software
+on your EV3, you'll need to connect to a network. Here are some resources to
+help you out.
+
+# Connecting to the internet
```
Comment (dlech) [inline docs/networking.md]: `<h1>` headings really should be title case as well.

File: docs/driver-overview.md (hunk 31 above)
Comment (dlech) [inline docs/driver-overview.md]: Actually, I didn't look to see if we are doing this on other existing pages.

File: docs/getting-started.md (hunk 29 above)
Comment (WasabiFan) [inline docs/getting-started.md]: The ID is composed of the group name and the subgroup name, so this is correct for the current behavior. The CI server checks complained at me after I changed the tutorial page because I had forgotten to update this link, but now it's happy that the link is valid.

File: docs/driver-overview.md (hunk 31 above)
Comment (WasabiFan) [inline docs/driver-overview.md]: I believe most of our pages use a sentence-case subtitle currently (e.g. <http://wasabifan.github.io/ev3dev.github.io/news/>). I personally like this style, for reasons similar to why many news organizations are now sentence-casing their titles: it seems more conversational.

File: docs/networking.md (hunk 32 above)
Comment (WasabiFan) [inline docs/networking.md]: Fixed
